win32/src/threadpool_win32.c

```diff
@@ -265,6 +286,59 @@ void threadpool_close(THANDLE(THREADPOOL) threadpool)
     }
 }
 
+/* threadpool_create_work_item */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_004: [ threadpool_create_work_item shall initialize the work_item_context to NULL, where NULL would be a failure indicator to the calling functions ] */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_005 : [ If threadpool is NULL, threadpool_create_work_item shall fail and return a NULL value. ] */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_006 : [ If work_function is NULL, threadpool_create_work_item shall fail and return a NULL value. ] */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_007 : [ Otherwise threadpool_create_work_item shall allocate a context where work_function, work_function_context, and ptp_work shall be saved. ] */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_008 : [ If any error occurs, threadpool_create_work_item shall fail and return a NULL value. ] */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_009 : [ threadpool_create_work_item shall create PTP_WORK by calling CreateThreadpoolWork to set the callback function as on_work_callback_v2 and pass the newly created context whenever the callback is called in the tp_environment. ] */
+
+/* Codes_SRS_THREADPOOL_WIN32_05_010 : [ If any error occurs, threadpool_create_work_item shall fail, free the newly created context and return a NULL value. ] */
+
+PVOID threadpool_create_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WORK_FUNCTION work_function, PVOID work_function_context)
+{
+    LogWarning("Printing %p %p %p", threadpool, work_function, work_function_context);

```

Comment:
>LogWarning("Printing %p %p %p", threadpool, work_function, work_function_context); [](http://example.com/codeflow?start=4&length=82)

this is fine, alternatively we will often just put lines like:

```
(void)threadpool;
(void)work_function;
(void)work_function_context;
```

For the otherwise unused variables